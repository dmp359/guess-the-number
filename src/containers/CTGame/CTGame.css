.window {
  font-size: 24px;
}

.instructions {
  padding: 8px;
}

.play-label {
  font-weight: bold;
  padding: 10px 0;
}
