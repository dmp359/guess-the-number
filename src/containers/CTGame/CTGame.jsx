import React, { Component } from 'react';
import { randInt } from 'utils/math';
import GuessBox from 'components/GuessBox/GuessBox.jsx';
import CTConfigGame from 'containers/CTConfigGame/CTConfigGame.jsx';
import PropTypes from 'prop-types';

import './CTGame.css';

/*
* Container for main gameplay
*/
class CTGame extends Component {
  constructor(props) {
    super(props);
    
    // Set default state and generate the mystery number
    const lower = props.lower || 1;
    const upper = props.upper || 10;
    this.state = { lower, upper, number: randInt(lower, upper), prev: 'none', msg: '' };

    this.onGuess = this.onGuess.bind(this);
    this.onReset = this.onReset.bind(this);
  }

  // Handle guess
  onGuess (guess) {
    const userGuess = guess || 0;
    const { number } = this.state;
    this.setState({ prev: guess });
    if (userGuess > number) {
      this.setState({ msg: 'Nope. Lower' });
    } else if (userGuess < number) {
      this.setState({ msg: 'Nope. Higher' });
    } else {
      this.setState({ msg: 'You got it!' });
    }
  };

  // Handle reset from config of new upper and lower value
  onReset(l, u) {
    this.setState({ lower: l, upper: u, number: randInt(l, u), prev: 'none', msg: '' });
  }

  render() {
    const { lower, upper, prev, msg } = this.state;

    // Only render a status (e.g. "too low!") if one is available
    let status;
    if (msg !== '') {
      status = (<div className='instructions'>{msg}</div>)
    }

    return (
      <div className='window'>
        <div className='play-label'>Play!</div>
        <div className='instructions'>
          Guess a number between {lower} and {upper}
        </div>
        <div className='instructions'>
          Previous guess was {prev}
        </div>
        {status}
        <GuessBox onSubmit={this.onGuess} />
        <CTConfigGame upper={upper} lower={lower} onReset={this.onReset} />
      </div>
    );
  }
}

CTGame.propTypes = {
  lower: PropTypes.number,
  upper: PropTypes.number,
}

export default CTGame;
