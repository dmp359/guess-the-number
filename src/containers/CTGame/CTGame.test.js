import React from 'react';
import { shallow } from 'enzyme';
import CTGame from './CTGame';

it('renders without crashing', () => {
  const component = shallow(<CTGame />);
  expect(component.debug()).toMatchSnapshot();
});

it('renders message if exists', () => {
  const component = shallow(<CTGame />);
  const str = 'test-message';
  expect(component.state('msg')).toEqual('');
  component.setState({ msg: str });
  expect(component.state('msg')).toEqual(str);
  component.update();
  expect(component.debug()).toMatchSnapshot();
});

it('handles a correct guess', () => {
  const component = shallow(<CTGame />);
  component.setState({ number: 5 }); // Mystery number is now 5
  component.instance().onGuess(5);
  component.update();
  expect(component.debug()).toMatchSnapshot();  
});

it('handles a low guess', () => {
  const component = shallow(<CTGame />);
  component.setState({ number: 5 }); // Mystery number is now 5
  component.instance().onGuess(4);
  component.update();
  expect(component.debug()).toMatchSnapshot();  
});

it('handles a high guess', () => {
  const component = shallow(<CTGame />);
  component.setState({ number: 5 }); // Mystery number is now 5
  component.instance().onGuess(6);
  component.update();
  expect(component.debug()).toMatchSnapshot();  
});

it('resets properly', () => {
  const component = shallow(<CTGame />);
  component.instance().onReset(0, 100);
  component.update();
  expect(component.debug()).toMatchSnapshot();  
});
