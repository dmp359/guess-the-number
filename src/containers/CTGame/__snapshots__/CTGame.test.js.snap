exports[`test handles a correct guess 1`] = `
"<div className=\"window\">
  <div className=\"play-label\">
    Play!
  </div>
  <div className=\"instructions\">
    Guess a number between 
    1
     and 
    10
  </div>
  <div className=\"instructions\">
    Previous guess was 
    5
  </div>
  <div className=\"instructions\">
    You got it!
  </div>
  <GuessBox onSubmit={[Function: bound onGuess]} />
  <CTConfigGame upper={10} lower={1} onReset={[Function: bound onReset]} />
</div>"
`;

exports[`test handles a high guess 1`] = `
"<div className=\"window\">
  <div className=\"play-label\">
    Play!
  </div>
  <div className=\"instructions\">
    Guess a number between 
    1
     and 
    10
  </div>
  <div className=\"instructions\">
    Previous guess was 
    6
  </div>
  <div className=\"instructions\">
    Nope. Lower
  </div>
  <GuessBox onSubmit={[Function: bound onGuess]} />
  <CTConfigGame upper={10} lower={1} onReset={[Function: bound onReset]} />
</div>"
`;

exports[`test handles a low guess 1`] = `
"<div className=\"window\">
  <div className=\"play-label\">
    Play!
  </div>
  <div className=\"instructions\">
    Guess a number between 
    1
     and 
    10
  </div>
  <div className=\"instructions\">
    Previous guess was 
    4
  </div>
  <div className=\"instructions\">
    Nope. Higher
  </div>
  <GuessBox onSubmit={[Function: bound onGuess]} />
  <CTConfigGame upper={10} lower={1} onReset={[Function: bound onReset]} />
</div>"
`;

exports[`test renders message if exists 1`] = `
"<div className=\"window\">
  <div className=\"play-label\">
    Play!
  </div>
  <div className=\"instructions\">
    Guess a number between 
    1
     and 
    10
  </div>
  <div className=\"instructions\">
    Previous guess was 
    none
  </div>
  <div className=\"instructions\">
    test-message
  </div>
  <GuessBox onSubmit={[Function: bound onGuess]} />
  <CTConfigGame upper={10} lower={1} onReset={[Function: bound onReset]} />
</div>"
`;

exports[`test renders without crashing 1`] = `
"<div className=\"window\">
  <div className=\"play-label\">
    Play!
  </div>
  <div className=\"instructions\">
    Guess a number between 
    1
     and 
    10
  </div>
  <div className=\"instructions\">
    Previous guess was 
    none
  </div>
  <GuessBox onSubmit={[Function: bound onGuess]} />
  <CTConfigGame upper={10} lower={1} onReset={[Function: bound onReset]} />
</div>"
`;

exports[`test resets properly 1`] = `
"<div className=\"window\">
  <div className=\"play-label\">
    Play!
  </div>
  <div className=\"instructions\">
    Guess a number between 
    0
     and 
    100
  </div>
  <div className=\"instructions\">
    Previous guess was 
    none
  </div>
  <GuessBox onSubmit={[Function: bound onGuess]} />
  <CTConfigGame upper={100} lower={0} onReset={[Function: bound onReset]} />
</div>"
`;
